mathics/builtin/system.py

```diff
@@ -389,3 +396,22 @@ class Version(Predefined):
 
     def evaluate(self, evaluation) -> String:
         return String(version_string.replace("\n", " "))
+
+
+class VersionNumber(Predefined):
+    r"""
+    <dl>
+      <dt>'$VersionNumber'
+      <dd>is a real number which gives the current Wolfram Language kernel vesion number \Mathics tries to be compatible with.
+    </dl>
+
+    >> $VersionNumber
+    = ...
+    """
+
+    name = "$VersionNumber"
+
+    def evaluate(self, evaluation) -> String:
+        # Make this be whatever the latest Mathematica release is,
+        # assuming we are trying to be compatible with this.
+        return Real(12.1)
```

Comment:
Here I really don't know enough about WolframLanguage to understand. 

While I would be the first to admit what we have in terms of state of advancement is 6.0 or even before that, thie purpose of this is to be able to use packages that state they need features of at least version X. So by putting this value high we don't shut off the possibility of important a package for this reason. Of course as we have seen there are lots of other reasons why things fail, and I would not be surprised if we don't have all of 6.0 covered as well. 

So are there features post 6.0 that we implement and therefore shouldn't attempt to install a packages that says it needs features after 6.0. If so, Ok, let's change. 

In fact, feel free to think about it and change the value to whatever you want and then merge.

Thanks.

Reply:
Usually, WL packages that look at the version, do it because they try to ensure that a certain feature is available. The current Mathics version supports things like Associations, which were not available in WL 6.0, but many other features like SparseArrays not.
So, I guess that v6.0 is a more or less safe choice and a reasonable goal for providing support.

Reply:
And of course, we can always override the version number by hand, to check if a certain package asking for a newer version could also work.

Reply:
I dunno much about older versions of Wolfram.

6.0 feels a bit too outdated, but considering the current state of mathics— I agree with @mmatera — 6.0 feels like a safe bet.